src/ragapp/backend/models/agent.py

```diff
@@ -0,0 +1,32 @@
+import re
+from datetime import datetime
+from typing import Dict
+
+from pydantic import BaseModel, Field, validator
+
+
+class ToolConfig(BaseModel):
+    enabled: bool = False
+    custom_prompt: str = ""
+    config: Dict = Field(default_factory=dict)
+
+
+class AgentConfig(BaseModel):
+    agent_id: str
+    name: str
+    role: str
+    system_prompt: str
+    tools: Dict[str, ToolConfig] = Field(default_factory=dict)
+    created_at: datetime = Field(default_factory=datetime.utcnow)
+
+    @validator("agent_id")
+    def validate_agent_id(cls, v):
+        if not re.match(r"^[a-zA-Z0-9_-]+$", v):
+            raise ValueError(
+                "agent_id must contain only alphanumeric characters, underscores, and hyphens"
```

Comment:
i guess this is the agent's name in the UI?
What error is shown in the UI if the name is wrong? Keep in mind that the user doesn't know what an `agent_id` is

Reply:
Great catch! This is redundant now because we now always generate uuid for the new agent.